lib/framework/procbot.ts

```diff
@@ -172,32 +176,39 @@ export class ProcBot {
 	 */
 	protected dispatchToAllEmitters(data: ServiceEmitRequest): Promise<ServiceEmitResponse[]> {
 		let results: ServiceEmitResponse[] = [];
+		let emitterPromises: Array<Promise<void>> = [];
 
 		// If there's not a context for a particular emmiter, it will result in a response
 		// with an error contained with in specifying as such. It is up to clients to determine
 		// whether this is an issue or not
-		return Promise.map(this.emitters, (emitter) => {
-			return emitter.sendData(data)
-			.then((result) => { results.push(result); })
-			.catch((error) => { results.push(error); });
-		}).return(results);
+		this.emitters.forEach((emitterEntry) => {
+			const emitter = <ServiceEmitter>emitterEntry.instance;
+			const promise = emitter.sendData(data)
+				.then((result) => { results.push(result); })
```

Comment:
Yep, it makes it more obvious (to me, anyway!) that they belong to the `const` declaration on the line above.